core/src/test/java/io/knotx/engine/core/KnotEngineTest.java

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * The code comes from https://github.com/tomaszmichalak/vertx-rx-map-reduce.
+ */
+package io.knotx.engine.core;
+
+import static io.knotx.engine.core.EntryLogTestHelper.verifyLogEntries;
+
+import com.google.common.collect.Lists;
+import io.knotx.engine.api.FragmentEvent;
+import io.knotx.engine.api.FragmentEvent.Status;
+import io.knotx.engine.api.FragmentEventResult;
+import io.knotx.engine.api.KnotFlow;
+import io.knotx.engine.api.KnotProcessingFatalException;
+import io.knotx.engine.api.TraceableKnotOptions;
+import io.knotx.engine.core.EntryLogTestHelper.Operation;
+import io.knotx.engine.core.impl.KnotEngineFactory;
+import io.knotx.fragment.Fragment;
+import io.knotx.knotengine.core.junit.MockKnotProxy;
+import io.knotx.server.api.context.ClientRequest;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.Vertx;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class KnotEngineTest {
+
+  @BeforeEach
+  void setUp() {
+    Assertions.assertTrue(System.getenv().containsKey("vertx.logger-delegate-factory-class-name"));
+  }
+
+  @Test
+  public void execute_whenEventWithNoKnot_expectEventWithUnprocessedStatus(
```

Comment:
How about using `@DisplayName("Expect unprocessed status when empty knot flow")`?
And then name method like `executeWithNoKnot` or smth like this?